plonky2/src/util/serialization/gate_serialization.rs

```diff
@@ -2,21 +2,31 @@ use plonky2_field::extension::Extendable;
 
 use crate::gates::gate::GateRef;
 use crate::hash::hash_types::RichField;
+use crate::plonk::circuit_data::CommonCircuitData;
 use crate::util::serialization::{Buffer, IoResult};
 
 pub trait GateSerializer<F: RichField + Extendable<D>, const D: usize> {
-    fn read_gate(&self, buf: &mut Buffer) -> IoResult<GateRef<F, D>>;
-    fn write_gate(&self, buf: &mut Vec<u8>, gate: &GateRef<F, D>) -> IoResult<()>;
+    fn read_gate(
+        &self,
+        buf: &mut Buffer,
+        common: &CommonCircuitData<F, D>,
```

Comment:
Is there a reason this isn't called `cd` as in the other functions?

Reply:
No! Will make the arg names uniform!

Reply:
Addressed, sorry about that!

Reply:
Np, thanks!